server/views/partials/expectedArrivalCardMacro.njk

```diff
@@ -11,17 +11,17 @@
             </aside>
             <div class="app-card__content">
                 <header class="app-card__header govuk-!-margin-bottom-4">
-                    <h2 class="app-card__title" data-qa="{{moveType}}-title-{{loopIndex}}">           
-                        {% if user.isReceptionUser and 
-                            (
-                                ['COURT', 'CUSTODY_SUITE'].includes(arrival.fromLocationType) 
-                                and not arrival.isCurrentPrisoner
-                            ) 
-                        %}
-                            <a href="/prisoners/{{ arrival.id }}/confirm-arrival" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a>
-                        {% else %}
-                            {{ arrival.lastName }}, {{ arrival.firstName }}
+                    <h2 class="app-card__title" data-qa="{{moveType}}-title-{{loopIndex}}">     
+                        {% if user.isReceptionUser %}
+                            {% if not arrival.isCurrentPrisoner %}
+                                    <a href="/prisoners/{{ arrival.id }}/confirm-arrival" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a>
+                            {% elif arrival.fromLocationType === 'PRISON' %}
+                                    <a href="/prisoners/{{ arrival.prisonNumber }}/check-transfer" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a> 
+                            {% else  %}
+                                    {{ arrival.lastName }}, {{ arrival.firstName }}
+                            {%  endif %}
```

Comment:
User must have Reception role.
Display prisoner name as a link if is an inward transfer from another prison or for non-transfers, prisoner is not a 'current' prisoner. 
Otherwise prisoner's name should not have a link.